bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/BookKeeper.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import org.apache.bookkeeper.client.BKException;
+import org.apache.bookkeeper.client.BookKeeperBuilderImpl;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+
+/**
+ * This is the entry point for BookKeeper client side API.
+ *
+ * @since 4.6
+ */
+public interface BookKeeper extends AutoCloseable {
+
+    /**
+     * Create a new builder which can be used to boot a new BookKeeper client. Additional contextual components can be
+     * injected to the client using {@link BookKeeperBuilder#with(java.lang.Object) } method.
+     *
+     * @param clientConfiguration the configuration for the client
+     * @return a builder
+     */
+    static BookKeeperBuilder newBuilder(final ClientConfiguration clientConfiguration) {
+        return new BookKeeperBuilderImpl(clientConfiguration);
+    }
+
+    /**
+     * Start the creation of a new ledger.
+     *
+     * @return a builder for the new ledger
+     */
+    CreateBuilder newCreateLedgerOp();
+
+    /**
+     * Open an existing ledger.
+     *
+     * @return a builder useful to create a readable handler for an existing ledger
+     */
+    OpenBuilder newOpenLedgerOp();
+
+    /**
+     * Delete an existing ledger.
+     *
+     * @return a builder useful to delete an existing ledger
+     */
+    DeleteBuilder newDeleteLedgerOp();
+
+    /**
+     * Close the client and release every resource.
+     *
+     * @throws BKException
+     * @throws InterruptedException
+     */
+    @Override
+    void close() throws BKException, InterruptedException;
```

Comment:
do we really need BKException on close?

Reply:
this was the legacy API, I cannot change the "close()" method of legacy BookKeeper impl

Reply:
if so, we need to do something here

a) I would suggest removing #close() from new API for now, we can use the #asyncClose() api in new API. 
b) if we want to keep such legacy method, I would suggest having a BKException as the parent class in api package. it is a bit strange from api package to reference legacy classes.

we can do either a) or b)

Reply:
I am for b). I  think we should implement AutoCloseable